server/src/test/java/org/opensearch/index/shard/RemoteStoreRefreshListenerTests.java

```diff
@@ -102,6 +103,16 @@ private void indexDocs(int startDocId, int numberOfDocs) throws IOException {
     public void tearDown() throws Exception {
         Directory storeDirectory = ((FilterDirectory) ((FilterDirectory) indexShard.store().directory()).getDelegate()).getDelegate();
         ((BaseDirectoryWrapper) storeDirectory).setCheckIndexOnClose(false);
+
+        for (ReferenceManager.RefreshListener refreshListener : indexShard.getEngine().config().getInternalRefreshListener()) {
+            if (refreshListener instanceof ReleasableRetryableRefreshListener) {
+                ((ReleasableRetryableRefreshListener) refreshListener).drainRefreshes();
+            }
+        }
+        if (remoteStoreRefreshListener != null) {
+            remoteStoreRefreshListener.drainRefreshes();
+        }
+
```

Comment:
why do we need this `drain` ? Ideally draining `ReleasableRetryableRefreshListener` should drain everything .

Reply:
Right, currently we create remoteStoreRefreshListener individually as well. I will create an issue to refactor this test to make the behaviour consistent.

Reply:
https://github.com/opensearch-project/OpenSearch/issues/12160